adr/0016-discarded-discovery-events.md

```diff
@@ -0,0 +1,33 @@
+# 16. Discarded Discovery Events
+
+Date: 2024-10-15
+
+## Status
+
+Accepted
+
+## Context
+
+The discovery workflow allows Trento Agent instances to publish their host system's information to the Trento Web's `collect` API operation. Once accepted, the event results in one or more commands that are dispatched to the ES/CQRS system.
+
+There are cases in which a discovery sent from a Trento Agent is not processed; they can be categorized as:
+1. The Trento Agent fails to connect to the Trento Web instance (network issues);
+2. The Trento Agent fails to authenticate with the Trento Web instance (authentication issues);
+3. The request payload is not in the expected shape, thus it cannot be mapped to a command (API contract issues);
+4. The discovery is rejected because of the current state in Trento.
```

Comment:
```suggestion
4. The discovery is rejected because of the current application state.
```

Reply:
I tried to come up with a third choice but I just think `state in Trento` is ok then. Let's go!